clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -645,6 +645,7 @@ struct StreamOperationEvaluator {
   SymbolRef StreamSym = nullptr;
   const StreamState *SS = nullptr;
   const CallExpr *CE = nullptr;
+  std::optional<CFGBlock::ConstCFGElementRef> ElemRef;
```

Comment:
Why is this declared here and not where the later assignment is?
As a rule of thumb, we should keep the lifetimes of objects as small as possible, and have their definitions as close to their uses as possible to enhance local reasoning.
Why do we need this optional anyway if we unconditionally unwrap it later?

Reply:
I'm not entirely sure whether I get your point correctly. This value needs to be stored when `Init` is called, and used in other calls (e.g. `bindReturnValue`). The reason why this is declared as an optional is that there is no default constructor for `ConstCFGElementRef`, so I used an optional to give an absent value (I guess it could have been a pointer like other ones)

Reply:
Ah I completely missed the context. nvm.